pilot/info/storageactivitymaps.py

```diff
@@ -11,20 +11,34 @@
 logger = logging.getLogger(__name__)
 
 
-sched2ddm_activity_map = {'es_events': 'pw',
-                          'es_failover': 'pw',
-                          'es_events_read': 'pr'}
+sched2ddm_activity_map_lan = {'es_events': 'write_lan',
+                              'es_failover': 'write_lan',
+                              'es_events_read': 'read_lan'}
 
 
-def get_ddm_activity(activity):
+sched2ddm_activity_map_wan = {'es_events': 'write_wan',
+                              'es_failover': 'write_wan',
+                              'es_events_read': 'read_wan'}
+
+
+def get_ddm_activity(activity, queuedata, ddmendpoint, ddm):
     """
     Map schedconf activities to corresponding ddm activities
 
     :param activity: schedconf activity as string
+    :param queuedata: Queue data info.
+    :param ddmendpoint: ddm endpoint.
+    :param ddm: ddm storage data.
     :return: ddm activity as string
     """
-    if activity in sched2ddm_activity_map:
-        ddm_activity = sched2ddm_activity_map[activity]
+    if activity in sched2ddm_activity_map_lan:
+        ddm_activity = sched2ddm_activity_map_lan[activity]
+        if queuedata and ddm_activity in queuedata.astorages and ddmendpoint and \
+                ddmendpoint in queuedata.astorages[ddm_activity] and ddm and \
```

Comment:
What is `queuedata.astorages[ddm_activity]`, why do you require it? normally it should not exist since `astorages` use `scheconfig_activity` as a key